internal/testutil/runmain.go

```diff
@@ -102,3 +106,35 @@ func (r *Runner) Run(env map[string]string, f func()) {
 	case <-done:
 	}
 }
+
+// RunNonInteractive runs the build binary until terminated or timeout has
+// been reached, and indicates successful execution on return.
+func (r *Runner) RunNonInteractive(env map[string]string, timeout time.Duration) bool {
+	if !r.Built() {
+		r.t.Error("Tried to run when binary not built.")
+		return false
+	}
+	environ := os.Environ()
+	for k, v := range env {
+		environ = append(environ, k+"="+v)
+	}
+
+	ctx, cancel := realContext.WithTimeout(realContext.Background(), timeout)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, r.bin)
+	cmd.Env = environ
+
+	out, err := cmd.Output()
+	r.Stdout = out
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			r.Stderr = exitErr.Stderr
+		}
+
+		// propagate this, or let callers decide?
```

Comment:
What if we return the error (and exit code)? Then, the caller has a bit more control than a bool.

Reply:
per cbro's suggestion, returning stdout/stderr and error value.